From dfd283e5c63bac205d6b404c3dac4a524689e29b Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Thu, 30 Dec 2021 17:10:27 +0200
Subject: [PATCH] [SW-36155] i2c_ind: prepare for using specific hdr

As a preparation for using specific RAS device
header, move C file ras implementation to gaudi,
and add include to device specific RAS header.

Change-Id: I22a6fe5fd2b99a22201405f7761173c82c9588a7
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index bf1c6cf..c887ad2 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -33,7 +33,6 @@
 #include "cmd_module.h"
 #include "i2cslave_indirect.h"
 #include "hl_i2c_slave_events.h"
-#include "rma_info.h"
 #include <spi_flash.h>
 #include <hl.h>
 
diff --git a/arch/gaudi/arm/hbm.c b/arch/gaudi/arm/hbm.c
index 5f6fe16..e50e351 100644
--- a/arch/gaudi/arm/hbm.c
+++ b/arch/gaudi/arm/hbm.c
@@ -5,17 +5,19 @@
  */
 
 #include <stddef.h>
+#include <stdint.h>
 
 #include <common/pci_ids.h>
 #include <gaudi_blocks.h>
+#include <preboot.h>
 #include <hl_i2c_indirect_protocol.h>
+#include <gaudi/i2c_indirect_protocol.h>
 #include <rma_info.h>
 #include <psoc_global_conf_regs.h>
 #include <gaudi/gaudi_fw_if.h>
 
 #include <types.h>
 #include <bitops.h>
-#include <preboot.h>
 #include <pll.h>
 #include <uart.h>
 #include <string.h>
diff --git a/arch/gaudi/arm/rma_info.c b/arch/gaudi/arm/rma_info.c
index b6cc63a..0bf3bb6 100644
--- a/arch/gaudi/arm/rma_info.c
+++ b/arch/gaudi/arm/rma_info.c
@@ -11,14 +11,15 @@
 #include <string.h>
 #include "uart.h"
 #include "boot_if.h"
+#include <preboot.h>
 #include <hl_i2c_indirect_protocol.h>
+#include <gaudi/i2c_indirect_protocol.h>
 #include "rma_info.h"
 #include "boot_info.h"
 #include "module_i2c_slave.h"
 #include "hl_i2c_slave_events.h"
 #include "i2cslave_indirect.h"
 #include "sha.h"
-#include <preboot.h>
 #include <hl_boot_desc.h> /* requires preboot.h for __packed */
 
 #define ROUND_UP4(n)	(((n) + sizeof(u32) - 1) & ~(sizeof(u32) - 1))
